### ISSUE TITLE
Absence of logic at GameObject.sendMessage()

### ISSUE THREAD

**Issue by wcobalt** [wcobalt/dengine#1]
Absence of logic when `INSTANCE_DESTROY` actually destroying an object, but e.g. `INSTANCE_CREATE` doesn’t create anything and so on. In other words, logic of messages lies across calling(higher) and called(lower) levels and higher level contains some logic (except sending messages) about lower levels. It's not right.

**Comment by wcobalt** [comment]
Logic was cleared, full comment from code: 

```
/*
 * So, sendMessage(semantic synonym - notify) is a method which responses on a message in proper way.
 * Actually, this method is called when on higher level, something will happen or was happened with the object.
 * Actions which necessary must be executed on higher level are executed there, but actions which COULD be executed as on higher
 * as on lower one, are executed on lower. So, developer of lower level sendMessage() logic, cannot rely on some actions
 * which could be executed somewhere on higher level. For instance:
 *
 * we have to destroy some GO (GameObject), since each GO is someone's child we must delete it from parent (parent, at this case
 * - is higher level, child is lower one). But before that we need to make child GO deleted itself. So, we notify it,
 * that it will be destroyed soon, and it cannot rely that IT's OWN children will be destroyed by someone, so it does it in it's own.
 * Like this, it cannot rely that IT's OWN components will be detached by someone, so it does itself.
 *
 * TL;DR: lower level must do work about itself, higher one about itself. Lower one cannot rely on higher one. sendMessage
 * is notification about something that was happened or will happen. It can be just notification (delegation of message to lower
 * levels) but can contain some logic as well.
 * */
```
Now it's clear what sendMessage must and must not do. Issue is closed

**Comment by wcobalt** [comment]
New full comment: /*
 * So, sendMessage(semantic synonym - notify) is a method which responses on a message in proper way.
 * Actually, this method is called when on higher level, something will happen or was happened with the object.
 * Actions which necessary must be executed on higher level are executed there, but actions which COULD be executed as on higher
 * as on lower one, are executed on lower. So, developer of lower level sendMessage() logic, cannot rely on some actions
 * which could be executed somewhere on higher level. For instance:
 *
 * we have to destroy some GO (GameObject), since each GO is someone's child, we must delete it from parent (parent, at this case
 * - is higher level, child is lower one). But before that we need to make child GO deleted itself. So, we notify it,
 * that it will be destroyed soon, and it cannot rely that IT's OWN children will be destroyed by someone, so it does it in it's own.
 * Like this, it cannot rely that IT's OWN components will be detached by someone, so it does itself.
 *
 * TL;DR: lower level must do work about itself, higher one about itself. Lower one cannot rely on higher one. sendMessage
 * is notification about something that was happened or will happen. It can be just notification (delegation of message to lower
 * levels) but can contain some logic as well.
 *
 * TL;DR in other words: this method and message system at all work as the following: higher level need to do something
 * with lower one, it does it on IT's level, then (IMPORTANT) it says to lower level: I did something with myself, and I'm sending
 * you a message in which it's described what I've done and how. If you need to do something too, ok, go ahead, read message
 * and do it. If you have lower levels, no problems, send according messages to them.
 *
 * Change #1: if there's interaction between objects (not necessary game ones) on the same layer, then the following scheme
 * works: objectA (one a method was called on) sends message to objectB (on the same layer). ObjectB somehow handles
 * message.
 *
 * Change #2: interaction with itself: if a GO does some stuff with itself and then it need to do some logic (e.g.
 * send a message too lower levels), it can send a message to itself firstly. The logic is not necessary propagation of
 * messages to lower levels. It can be any logic. BUT, if this logic is used once in the GO class and it isn't propagation
 * it is recommended to exclude it logic to methods where it is needed. So, if it's needed e.g. to delete a child and
 * then propagate according message to lower levels, propagation goes in sendMessage(), but directly deletion goes in
 * method-invoker.
 * */
